src/DOM/File/FileList.purs

```diff
@@ -0,0 +1,25 @@
+module DOM.File.FileList
+  ( length
+  , item
+  , toArray
+  ) where
+
+import Prelude ((-))
+import Data.Array as Array
+import Data.Maybe (Maybe())
+import Data.Nullable (Nullable(), toMaybe)
+import DOM.File.Types (FileList(), File())
+
+-- | Number of files in the `FileList` object.
+foreign import length :: FileList -> Int
+
+-- | Get `File` at the certain position
+foreign import itemImpl :: Int -> FileList -> Nullable File
+
+item :: Int -> FileList -> Maybe File
+item i l = toMaybe (itemImpl i l)
```

Comment:
Likewise this actually.

They're both definitely useful, but this library is supposed to be providing a low level unopinionated interface, so I think the `Nullable File` version is fine.